### ISSUE TITLE
color problem on Vista

### ISSUE THREAD

**Issue by pphaneuf** [quadra-game/quadra#62]

**Comment by pphaneuf** [comment]
-&gt; 4 out of 5 times i start quadra, the graphics are extrem low quality,
looks like 8 colors ega ;). Can't really reproduce the case, when it was
fine. Quadra always starts in fullscreen, no window-mode with
&quot;-nofullscreen&quot; possible or switching with alt-enter.

-&gt; At the moment i start a game (single player, clicking start when joining
a internet-server from nx or replay a game from the highscore list)
quadra.exe crashes.

Windows Vista Business German, SP-1 here.

I'll try to make enough space on my hd to install the directx-sdk for
debugging purposes, perhaps then i can help more than &quot;it doesn't work&quot; ;)

**Comment by pphaneuf** [comment]
I have seen the issue with the wrong colors a few times here as well (though nowhere 
near 4/5 of the time) on Vista Home Premium. Note that this version doesn't have a 
windowed mode; that will only be available in the SDL version.

Does the crash only occur when the colors are wrong or is it all the time?

Another note: the correct Direct X SDK version is the one from August 2007. Later 
versions have removed some stuff that we use. Maybe that only applies at build time 
(i.e. it's possible that the latest version could help debugging anyway) but if 
you're installing it just for Quadra, it might be worth your while to dig a little 
for that specific version.

**Comment by pphaneuf** [comment]
Assigning to our resident Windows expert. ;-)

**Comment by pphaneuf** [comment]
Ok, when i switch my to 16bit color depth, quadra looks fine. Even if i change back
to 32 bit the next time i start quadra it seems to be able to switch for itself...

The problem with the crash on game start remains. Neither "Run as admin" nor WinXP
compatibility mode changes seems to help.

I've download the DirectX-SDK, but March 08. You are right, quadra doesn't
compile/link with this newer version. Have to find the older version, but it seems
every page redirects to the actual version...

**Comment by pphaneuf** [comment]
That's really annoying, is there anything we can do? Stéphane, have you got
reproduction on this?

**Comment by pphaneuf** [comment]
Tino, if you've got a development environment set up (minus the problematic DirectX SDK version), maybe 
you could try the SDL branch?

I'm not sure if this would be a conclusive way of resolving that bug ("oh, if you have Vista, wait for 1.4, in a 
few years"), but I'm curious to know if the SDL version has the same problem. Since it seems to be heavily 
buggy on your system, you'd be the best test case. :-)

Also, have you tried Quadra 1.1.8 (the last released version)? Does it have the same problems? This could help 
us isolate the problem better...

Thanks!

**Comment by pphaneuf** [comment]
I think that if we don't get this right, we might just end up getting pushed soon for
"why doesn't it work in Vista"... Since I'd like this one to be the last 1.2, if
possible, if we could bury this, that'd be awesome.

Tino: If you manage to build trunk on your machine (resolving your DirectX SDK
problems), something that would be EXTREMELY helpful was if you could help pinpoint
the revision that introduced the problem. Basically, you look at
http://code.google.com/p/quadra/source/list?path=/trunk/quadra/ and go update to one
of those older revision, test if it's working okay or not, then go forward or
backward, try again, etc...

Either you'll hit a point so far in the past that it won't compile anymore (knowing
the oldest one that you could build would still be helpful) but you'll still have the
issue, or (preferably!) you'll find a revision which is okay, and where the one right
after isn't (introducing the bug). That latter case would probably make it super-easy
to fix. :-)

Thanks!

**Comment by pphaneuf** [comment]
Oh, Tino, just so you know, 1.1.8 (which you said was okay, I think?) seems to be
r175, so you shouldn't need to go further back than that.

**Comment by pphaneuf** [comment]
I separated out the second point in the description to its own issue #80.

**Comment by pphaneuf** [comment]
Rémi and I did some research on the subject, and discussed it a little with Stéphane...

What we found is that it seems to be a common problem with DirectDraw games on Vista,
and it seems to be related with other programs trying to display something, and while
they fail (since the DirectDraw program is fullscreen), their attempt seems to
disrupts the palette. Descriptions of the problem seemed to associate it with
programs such as anti-virus software, Google Desktop, things of that nature.

While Stéphane did experience the problem, he did so only _twice_, having started the
program untold number of times. So we basically don't have reproduction.

Since we don't have reproduction (although I suggested to Stéphane he could try
installing Google Desktop and seeing if it does anything to Quadra), and this seems
to be a well-known problem of DirectDraw games on Vista, we agreed that we would
probably ship with this problem, unfortunately.

Since the SDL version does not (normally) use DirectDraw (it uses the GDI instead,
which is now faster than DirectDraw on DirectX 9 and newer when on 2D hardware, and
is in any case "sufficiently fast" for a game like Quadra), it should not have the
same problem, so that might be a way out for you?

Any new information is welcome, and could change our decision in this matter. If we
could do something about it, we would, sorry!
